vms/platformvm/camino_service.go

```diff
@@ -416,6 +408,14 @@ func (s *CaminoService) Spend(_ *http.Request, args *SpendArgs, response *SpendR
 	s.vm.ctx.Lock.Lock()
 	defer s.vm.ctx.Lock.Unlock()
 
+	privKeys, err := s.getFakeKeys(&args.JSONFromAddrs)
+	if err != nil {
+		return err
+	}
+	if len(privKeys) == 0 {
+		return errNoKeys
+	}
+
```

Comment:
because getFakeKeys actually uses common context (deeper in the stack call), so it must be protected against racing.
We didn't notice it in original dev branch PRs.